PPos pos
ControlInput in
float32 weight
int16 collision
int16 num_particles
int16 parent_id
